### ISSUE TITLE
Issue with docker run instructions

### ISSUE THREAD

**Issue by LivingWithHippos** [coldfix/certbot-dns-netcup#9]
I tried following the docker instruction but I had the error `certbot: error: unrecognized arguments: certbot`. Apparently certbot is executed already so I had to remove it and it almost worked

```shell
docker run --rm \
  -v /var/lib/letsencrypt:/var/lib/letsencrypt \
  -v /etc/letsencrypt:/etc/letsencrypt \
  --cap-drop=all \
  coldfix/certbot-dns-netcup certonly \
    --authenticator dns-netcup \
    --dns-netcup-propagation-seconds 900 \
    --dns-netcup-credentials /var/lib/letsencrypt/netcup_credentials.ini \
    --no-self-upgrade \
    --keep-until-expiring --non-interactive --expand \
    --server https://acme-v02.api.letsencrypt.org/directory \
    -d example.com -d '*.example.com'
```

after that I still had issues and the message `You should register before running non-interactively, or provide --agree-tos and --email <email_address> flags.`

```shell
docker run --rm \
  -v /var/lib/letsencrypt:/var/lib/letsencrypt \
  -v /etc/letsencrypt:/etc/letsencrypt \
  --cap-drop=all \
  coldfix/certbot-dns-netcup certonly \
    --authenticator dns-netcup \
    --dns-netcup-propagation-seconds 900 \
    --dns-netcup-credentials /var/lib/letsencrypt/netcup_credentials.ini \
    --no-self-upgrade \
    --keep-until-expiring --non-interactive --expand \
    --server https://acme-v02.api.letsencrypt.org/directory \
    --agree-tos \
    --email address@email.com \
    -d example.com -d '*.example.com'
```

This made it work

**Comment by coldfix** [comment]
Hi,

thanks for reporting this. The first issue is also mentioned in the README: the executable name `certbot` was part of the ENTRYPOINT in older certbot versions, so it had to be omitted from the command line.

It's weird that you get this behaviour now, because I just pulled the most recent version, and it seems to work exactly as stated in the README, i.e. *with* the `certbot` executable name. Which version are you running? My best theory is that you build the docker image locally, and had an old version of `certbot/certbot` already cached. In this case, try first updating with `docker pull certbot/certbot` and then rebuild.

Good point as for the `--agree-tos` and `--email` options, these are definitely needed (and I'm also using them in my own setup).

**Comment by LivingWithHippos** [comment]
You say `docker pull certbot/certbot` but your docs say `docker pull coldfix/certbot-dns-netcup` so that's probably the issue ;D

**Comment by coldfix** [comment]
There is two ways to get an image:

1. `docker pull coldfix/certbot-dns-netcup` to directly get the latest version from docker hub.
2. `docker build -t coldfix/certbot-dns-netcup .` to locally build a new image from the checked out repository. This image is based on `certbot/certbot`. A pitfall with this approach is that it can reuse a cached version of the `certbot/certbot` image that was downloaded a long time ago. That means, if you have previously used certbot, and now build the plugin here, it will just use the old certbot version. In general, this might work fine, except for some differences like the one you observed (and also differences regarding prefixes of the option names). To fix this, one can execute `docker pull certbot/certbot` before the `docker build` so that the new image will be based on a different certbot version.

I would be curious if you followed method 1 or method 2, because case 2 would be a natural explanation for the observed behaviour, that doesn't require further fixes (except maybe pinning a minimal certbot version), while there would be something weird going on here, if you used the first method.

I tried both ways (pull the latest `coldfix/certbot-dns-netcup` or build from scratch), and in both cases the `certbot` executable name seems to be necessary for me.

**Comment by LivingWithHippos** [comment]
sorry I forgot to specify, I used `docker pull coldfix/certbot-dns-netcup` and portainer is showing `coldfix/certbot-dns-netcup@sha256:f4361ea54dde747478f997766742114509eeaf592ca7e287f37b6f884f0a0753` as image. I ran it yesterday for the first time so no cache.

**Comment by coldfix** [comment]
Turns out I was wrong. Somehow must have tested a different image... I just don't know how I even had a version with different entrypoint behaviour (maybe I was using an old cached version for some reason?).

Anyway, I fixed it in `v1.1.0` to not using `certbot` as `ENTRYPOINT`, i.e. having to manually specify the executable name it as it was described in the readme.

Many thanks for your patience.
